### DISCUSSION TITLE
How to redirect all the incoming requests for multiple microservices?

### DISCUSSION THREAD

**Discussion by AShaker01** [ThreeMammals/Ocelot#1841] [Q&A] [answered]
Hello Folks,
I have a scenario that want to apply but I don't know how.

I've multiple Services [for example KM and NLP]
KM is hosted on localhost:3000 and NLP hosted on localhost:4000

What I want to achieve is the following:

If I get any requests like:
- `/api/v1/KM/{anything}` this should forward to `localhost:3000/api/v1/KM/{anything}`
- `/api/v1/NLP/{anything}` this should forward to `localhost:4000/api/v1/NLP/{anything}`

another example I can get is something like that
- `/api/v1/KM/Customers/{CustomerId}` also this should be redirected to `localhost:3000/api/v1/KM/Customers/{CustomerId}`

How can I do something like that?

So I need to redirect the URL as is and have to take into consideration the HTTP action method [GET, PUT, etc..]

Is that possible?

I tried something like that, but doesn't work as expected.
```json
"UpstreamPathTemplate": "/api/v1/KM/{everything}",
"DownstreamPathTemplate": "/api/v1/KM/{everything}",
"DownstreamScheme": "https",
"DownstreamHostAndPorts": [
  {
    "Host": "localhost",
    "Port": 3000
  }
```

**Comment by raman-m** [comment]
@ks1990cn Please help!

@AShaker01 
Your route/URL `/api/v1/KM/Customers/{CustomerId}` is a part of `/api/v1/KM/{anything}`!
I don't see any reason to define 2 routes based on your description.

I see a sense only if the hosts are different.

**Reply by raman-m** [reply]
@AShaker01 Where are you?

**Comment by ks1990cn** [comment]
@AShaker01 Hi, You need your route like this in Ocelot.json

Here is doc https://ocelot.readthedocs.io/en/latest/features/routing.html

Make sure you are not conflicting with paths.

More help - https://www.youtube.com/watch?v=hlUGZ6Hmv6s

What do you exactly mean when you say `I tried something like that, but doesn't work as expected` ?

```
  "Routes": [
    {
      "DownstreamHostAndPorts": [
        {
          "Host": "localhost",
          "Port": 3000
        }
      ],
      //Your paths here KM
      "UpstreamPathTemplate": "/hostname",
      "UpstreamHttpMethod": [ "GET" ],
      "DownstreamPathTemplate": "/WeatherForecast/hostname",
      "DownstreamScheme": "http"

    },
    {
      "DownstreamHostAndPorts": [
        {
          "Host": "localhost",
          "Port": 4000
        }
      ],
      //Your paths here NLM
      "UpstreamPathTemplate": "/hostname",
      "UpstreamHttpMethod": [ "GET" ],
      "DownstreamPathTemplate": "/WeatherForecast/hostname",
      "DownstreamScheme": "http"
    }
  ]
```

**Reply by raman-m** [reply]
Tanmay, I love when you joking. 🤣

**Reply by ks1990cn** [reply]
Hahahaha!! 😆

**Reply by raman-m** [reply]
In both your routes UpstreamPathTemplate props are the same, and they don't match author's one!

As I wrote above, author asked very strange question, because 3rd route can be a part of previous one.
Well... I expect the author's silence...

**Reply by ks1990cn** [reply]
Yeah I added comment above it, expecting him to write his own🤣

**Reply by ks1990cn** [reply]
I have already shared lot of docs for him with tutorial on YT.

Hopefully we are replying him what he wants🤣🤣🤣

**Reply by ks1990cn** [reply]
I think he just want different upstream to connect with his different downstream

**Comment by AShaker01** [comment, marked as answer]
Using everything in the route working as expected. 
the issue was that I've a route like/api/products
and this should be mapping to products controller get method.
but when I've defined the upstream route likes /api/products/everything.
all the routes worked except that route /api/products

so i ended up by defining two routes
one specific for the api/products
and one generic for all other routes api/products/{everything}

that's way in the first glance i thought that everything doesn't work :)

thanks for everyone's support and jokes 😁

**Reply by raman-m** [reply]
> one specific for the api/products
> and one generic for all other routes api/products/{everything}

Why not to define one general route?
- `api/{products}`, here `{products}` is like `{everything}` placeholder

Why 2 routes? I cannot get you!..
